analyzer/decrypt_crypto.cc

```diff
@@ -200,24 +200,22 @@ std::vector<uint8_t> calculate_nonce(std::vector<uint8_t> client_iv, uint64_t pa
 Function that calls the AEAD decryption routine, and returns the
 decrypted data
 */
-std::vector<uint8_t> decrypt(std::vector<uint8_t> client_key, std::vector<uint8_t> encrypted_packet,
-                             uint64_t payload_length, DecryptionInformation decryptInfo)
+
+hilti::rt::Bytes decrypt(const std::vector<uint8_t>& client_key,
+                         const std::vector<uint8_t>& encrypted_packet, uint64_t payload_length,
+                         const DecryptionInformation& decryptInfo)
 	{
 	int out, out2, res;
-	std::vector<uint8_t> encrypted_payload(
-		encrypted_packet.begin() + decryptInfo.protected_header.size(),
+	const uint8_t* encrypted_payload = &encrypted_packet[decryptInfo.unprotected_header.size()];
+	int encrypted_payload_size = payload_length - decryptInfo.packet_number_length -
+	                             AEAD_TAG_LENGTH;
```

Comment:
I'd prefix this with a check for underflows (unsure whether that would be due to input or programming error, so it should either throw or assert).

Reply:
That's already done one layer up, but I'll place one here, too.